### PR TITLE
light-dark-theme

### PR DESCRIPTION
**Pull request by 47codemonkey**
Kindly check my monkey code. Wait for your comments and ready to implement changes.

### CODE REVIEW

**Comment by 47codemonkey** [comment]
"@rbnd/react-dark-mode": "^2.0.1",

pls ignore it. I will uninstall it on my next pr

File: src/components/lightDarkTheme/LightDarkTheme.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { useState, useEffect, useCallback } from 'react';
+
+export const LightDarkThemeToggle = () => {
+  const defaultTheme = 'light';
```
Comment (dmytro-honch) [inline src/components/lightDarkTheme/LightDarkTheme.tsx]:
``` JS
const defaultTheme = window.matchMedia('(prefers-color-scheme: dark)')?.matches ? 'dark' : 'light';
```

Comment (47codemonkey) [reply src/components/lightDarkTheme/LightDarkTheme.tsx]: changed

File: src/components/lightDarkTheme/LightDarkTheme.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import { useState, useEffect, useCallback } from 'react';
+
+export const LightDarkThemeToggle = () => {
+  const defaultTheme = window.matchMedia('(prefers-color-scheme: dark)')?.matches ? 'dark' : 'light';
```
Comment (dmytro-honch) [inline src/components/lightDarkTheme/LightDarkTheme.tsx]: Good practice is move such string as "dark" and "light" in constants. The same as for options for dropdowns, labels and such stuff.  
Let's do it later.

File: src/components/lightDarkTheme/LightDarkTheme.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  }, [theme]);
+
+  const applyThemeStyles = () => {
+    const body = document.body;
+    if (theme === 'light') {
+      body.setAttribute('data-theme', 'light');
+    } else {
+      body.setAttribute('data-theme', 'dark');
+    }
+  };
```
Comment (dmytro-honch) [inline src/components/lightDarkTheme/LightDarkTheme.tsx]: All this logic should be excluded in ./hook.ts file for this component.

Comment (dmytro-honch) [inline src/components/lightDarkTheme/LightDarkTheme.tsx]: All this logic should be excluded as custom hook in ./hook.ts file for this component.

**Review by dmytro-honch — APPROVED** [review]

File: src/components/lightDarkTheme/LightDarkTheme.tsx (hunk 3 above)
Comment (47codemonkey) [reply src/components/lightDarkTheme/LightDarkTheme.tsx]: Done

File: src/components/lightDarkTheme/LightDarkTheme.tsx (hunk 2 above)
Comment (47codemonkey) [reply src/components/lightDarkTheme/LightDarkTheme.tsx]: have some issues with path. Kindly check my last commit "some architecture changes"

File: src/components/lightDarkTheme/hook.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+      applyThemeStyles();
+    }, [theme]);
+    
+    const applyThemeStyles = () => {
+      const body = document.body;
+      if (theme === light) {
+        body.setAttribute('data-theme', light);
+      } else {
+        body.setAttribute('data-theme', dark);
+      }
```
Comment (dmytro-honch) [inline src/components/lightDarkTheme/hook.ts]: Let's avoid `else` everywhere :)

``` JS
if (theme === light) {
  body.setAttribute('data-theme', light);
  return;
}

body.setAttribute('data-theme', dark);
```

File: src/components/lightDarkTheme/hook.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import { useState, useEffect, useCallback } from 'react';
+import { dark, light } from 'src/constants';
+
+
+export const useLightDarkTheme = () => {
+
+    const defaultTheme = window.matchMedia('(prefers-color-scheme: dark)')?.matches ? 'dark' : 'light';
+    const [theme, setTheme] = useState(localStorage.getItem('theme') || defaultTheme);
```
Comment (dmytro-honch) [inline src/components/lightDarkTheme/hook.ts:8]: could be case, when user opens App for first time and their browser doesn't support `window.matchMedia('(prefers-color-scheme: dark)')`. In this case theme === undefined. Better to set there third optional case:
``` JS
      const [theme, setTheme] = useState(localStorage.getItem('theme') || defaultTheme || THEMES.light);
```

File: src/components/lightDarkTheme/hook.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
+import { useState, useEffect, useCallback } from 'react';
+import { dark, light } from 'src/constants';
+
+
+export const useLightDarkTheme = () => {
+
+    const defaultTheme = window.matchMedia('(prefers-color-scheme: dark)')?.matches ? 'dark' : 'light';
```
Comment (dmytro-honch) [inline src/components/lightDarkTheme/hook.ts]:
``` JS
?.matches ? THEMES.dark : THEMES.light;
```

File: src/components/lightDarkTheme/hook.ts (hunk 4 above)
Comment (47codemonkey) [reply src/components/lightDarkTheme/hook.ts]: changed

File: src/components/lightDarkTheme/hook.ts (hunk 6 above)
Comment (47codemonkey) [reply src/components/lightDarkTheme/hook.ts]: changed

File: src/components/lightDarkTheme/hook.ts [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+  }, [theme]);
+
+  useEffect(() => {
+    applyThemeStyles(theme);
+  }, [theme]);
+
+  const applyThemeStyles = (currentTheme) => {
+    const body = document.body;
+    body.setAttribute('data-theme', currentTheme);
+  };
```
Comment (dmytro-honch) [inline src/components/lightDarkTheme/hook.ts]: let's keep flow for compilation:
1. applyThemeStyles 
2. useEffect
3. handleToggleTheme

B/c now you call function (applyThemeStyles) before it's expression.

File: src/components/lightDarkTheme/hook.ts [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+//       const body = document.body;
+//       if (theme === THEMES.light) {
+//         body.setAttribute('data-theme', THEMES.light);
+//         return
+//       } 
+//       body.setAttribute('data-theme', THEMES.dark);
+//     };
+    
+//     return { toggleTheme, applyThemeStyles };
+// };
```
Comment (dmytro-honch) [inline src/components/lightDarkTheme/hook.ts]: Are you collecting it? Remove :)

File: src/constants/lightDarkTheme.ts [hunk 9]
```diff
@@ -0,0 +1,4 @@
+export const THEMES = {
+    light: 'light',
+    dark: 'dark',
+  };
```
Comment (dmytro-honch) [inline src/constants/lightDarkTheme.ts]: themes.tsx

File: src/components/lightDarkTheme/hook.ts (hunk 8 above)
Comment (47codemonkey) [reply src/components/lightDarkTheme/hook.ts]: sorry)

File: src/components/lightDarkTheme/hook.ts (hunk 7 above)
Comment (47codemonkey) [reply src/components/lightDarkTheme/hook.ts]: okay

File: src/constants/lightDarkTheme.ts (hunk 9 above)
Comment (47codemonkey) [reply src/constants/lightDarkTheme.ts]: done

**Pull request merged**